vmdb/app/models/job_proxy_dispatcher.rb

```diff
@@ -208,7 +212,14 @@ def busy_resources_for_embedded_scanning
     $log.debug("MIQ(JobProxyDispatcher.busy_resources_for_embedded_scanning) Initializing busy_resources_for_embedding_scanning hash")
     @busy_resources_for_embedded_scanning_hash ||= {}
 
-    vms_in_embedded_scanning = Job.find(:all, :conditions => ["dispatch_status = ? AND state != ? AND agent_class = ? AND target_class = ?", "active", "finished", "MiqServer", "VmOrTemplate"], :select => "target_id").collect {|ji| ji.target_id}.compact.uniq
+    vms_in_embedded_scanning =
+      Job.where(:dispatch_status => "active")
+        .where(:agent_class      => "MiqServer")
+        .where(:target_class     => "VmOrTemplate")
+        .where("state != ?", "finished")
+        .pluck("DISTINCT target_id")
```

Comment:
In 3.2, you can do this:

```ruby
irb(main):010:0> MiqProductFeature.pluck("DISTINCT parent_id")
  SQL (1.7ms)  SELECT DISTINCT parent_id FROM "miq_product_features"
```

```ruby
irb(main):011:0> MiqProductFeature.uniq.pluck("parent_id")
  SQL (0.9ms)  SELECT DISTINCT parent_id FROM "miq_product_features"
```

Rails 4.2 adds .distinct:
```ruby
irb(main):001:0> MiqProductFeature.uniq.pluck("parent_id")
   (0.7ms)  SELECT DISTINCT "miq_product_features"."parent_id" FROM "miq_product_features"

irb(main):002:0> MiqProductFeature.distinct.pluck("parent_id")
   (0.7ms)  SELECT DISTINCT "miq_product_features"."parent_id" FROM "miq_product_features"
```

http://api.rubyonrails.org/classes/ActiveRecord/QueryMethods.html#method-i-distinct

.uniq and .distinct are aliases in rails 4.2.

I think I'll change the call to .uniq since .distinct doesn't exist in rails 3.2